cmd/cdi-controller/controller.go

```diff
@@ -287,6 +288,9 @@ func start() {
 
 	klog.V(1).Infoln("created cdi controllers")
 
+	// Setup Collectors
+	monitoring.SetupPersistentVolumesCollector(client)
```

Comment:
At every Prometheus scrape, which by default should be every 30 seconds

Reply:
There should be a more controller-runtime-friendly way to register these and benefit from a cached client

Reply:
Updated to use informers

Reply:
You should be able to use the controller runtime cached client, which should be usable once the manager is created
(mgr.GetClient())

Reply:
what would be the difference between the 2 approaches?

Reply:
So the controller runtime client already encapsulates these things for you, there should be no reason to roll out informers
on your own

Reply:
Am I missing some complications of this? It's just that it's weird to introduce an informer when the framework gives you a cached client

Reply:
Ah, sorry, I forgot to answer earlier. [controller-runtime uses SharedIndexInformer](https://github.com/kubernetes-sigs/controller-runtime/blob/9ed8996ae17645d1adc0b30ae186594333b5691a/pkg/cache/internal/informers_map.go#L209). I prefer to use the informer directly because it reduces monitoring dependability on controller-runtime or other specificities. Informer works with any client

Reply:
Since we only use controller runtime in CDI, I don't think that is a problem, we really prefer you use the client from the runtime.